core/GlobalState.cc

```diff
@@ -1152,6 +1152,7 @@ unique_ptr<GlobalState> GlobalState::deepCopy(bool keepId) const {
     result->deepCloneHistory = this->deepCloneHistory;
     result->deepCloneHistory.emplace_back(DeepCloneHistoryEntry{this->globalStateId, namesUsed()});
 
+    result->wasModified_ = this->wasModified_;
```

Comment:
ouch. Does it mean that all code that uses this boolean is wrong? :-(

Reply:
This was not a big problem on non-plugin runs. The only code that uses this is the logic for deciding whether to write the cache to disk after indexing. on big typecheck runs, where caching is useful, it's probably very very hard to get into a situation where we forget to write the cache. even if we do forget, it's not the end of the world either.